GitUI/CommandsDialogs/FormResolveConflicts.cs

```diff
@@ -1243,6 +1256,28 @@ private void OpenMergetool_Click(object sender, EventArgs e)
             OpenMergeTool();
         }
 
+        private void customMergetool_Click(object sender, EventArgs e)
+        {
+            if (sender == customMergetool)
+            {
+                // Cancel dropdown manually if "main menu" is selected
+                customMergetool.HideDropDown();
+                ConflictedFilesContextMenu.Hide();
```

Comment:
Shouldn't we exit here? `customTool` will be `null`, won't it?

Reply:
if customTool is null or empty, the default mergetool will be used so this works fine.

In this case it is possible to select both the top level "Open in mergetool" and the menu. Maybe mostly used while the list of tools is building (can takes ten seconds on PC with virus programs). If top is selected, the menus must be closed directly. An alternative is to ignore the top level selection once there is a dropdown created.

For the custom difftool action in revdiff, top level has no effect as there is a separate menu item.

![image](https://user-images.githubusercontent.com/6248932/101375703-02f5e100-38b0-11eb-9e7d-45c364523eed.png)

A side discussion, I would like to use custom difftool in more scenarios (like History, diff any two files), then adding a second menu item will duplicating several items and the top level with dropdown should be considered.

Reply:
> A side discussion, I would like to use custom difftool in more scenarios (like History, diff any two files)

FormHistory is missing too much. And I do not see a real difference to FormBrowse. That's why I would not maintain FormHistory but replace it with a variant of FormBrowse as proposed in #7598, (which has become even easier since the history is shown in a dedicated GE instance).

Reply:
History discussed in separate issue

If other forms are changed is also not in this PR, except that I would like to use the same behavior if changed elsewhere: action on the main menu and submenu